mcm/tests/campaign/test_campaign.py

```diff
@@ -0,0 +1,268 @@
+            - restapi/campaigns/get_cmsDrivers/<string:campaign_id>
+        """
+        endpoint = f"restapi/campaigns/get_cmsDrivers/{mockup.get('prepid')}"
+        return self.mcm._get(endpoint)
+
+    def inspect(self, mockup: dict):
+        """
+        Inspect all the requests in a given campaign.
+        Related to test the endpoint:
+            - restapi/campaigns/inspect/<string:campaign_id>
+        """
+        endpoint = f"restapi/campaigns/inspect/{mockup.get('prepid')}"
+        return self.mcm._get(endpoint)
+
+    def status(self, mockup: dict):
+        """
+        Toggles the status for a campaign.
+        Related to test the endpoint:
+            - restapi/campaigns/status/<string:campaign_id>
+        """
+        endpoint = f"restapi/campaigns/status/{mockup.get('prepid')}"
+        return self.mcm._get(endpoint)
+
+
+class TestCampaign(EntityTest):
+    """
+    Test the campaign API.
+    """
+
+    @property
+    def entity_api(self) -> CampaignAPI:
+        mcm: McM = McM(config=config, role=Roles.ADMINISTRATOR)
+        return CampaignAPI(mcm)
+
+    def test_create(self):
+        # Invalid name
+        invalid_mockup = self.entity_api.mockup()
+        invalid_mockup["prepid"] = "Run3Campaign#1"
+        content, response = self.entity_api.create(invalid_mockup)
+
+        assert response.status_code == 200
+        assert content["results"] == False
+
+        # Valid campaign
+        valid_mockup = self.entity_api.mockup()
+        content, response = self.entity_api.create(valid_mockup)
+
+        assert response.status_code == 200
+        assert content["results"] == True
+
+    def test_retrieve(self):
+        # The campaign exists
+        mockup = self.entity_api.mockup()
+        self.entity_api.create(mockup)
+        retrieved = self.entity_api.retrieve(mockup)
+        del mockup["history"]
+        del retrieved["history"]
+        del retrieved["_rev"]
+
+        assert mockup == retrieved
+
+        # The campaign doesn't exists
+        not_exists = self.entity_api.retrieve(self.entity_api.mockup())
+        assert not_exists == None
+
+    def test_update(self):
+        # The campaign is updated to a valid status
+        example = self.entity_api.example()
+        example["cmssw_release"] = "CMSSW_14_0_0"
+        content, response = self.entity_api.update(example)
+        retrieved = self.entity_api.retrieve(example)
+
+        assert response.status_code == 200
+        assert content["results"] == True
+        assert example["_rev"] != retrieved["_rev"]
+        assert retrieved["cmssw_release"] == "CMSSW_14_0_0"
+
+        # Update a campaign that doesn't exist.
+        example = self.entity_api.example()
+        example["prepid"] = "Run3Campaign#2"
+        content, response = self.entity_api.update(example)
+
+        assert response.status_code == 200
+        assert content["results"] == False
+
+    @pytest.mark.skip("Takes too much time to compute")
```

Comment:
How comes it's so slow? I think the test should be implemented but marked as [`slow`](https://docs.pytest.org/en/7.1.x/how-to/mark.html#mark).

Reply:
I have included a test, need to check if works properly when deploying into test env